File: 2015/day05.py

nice = sum( 1 for ln in open('day05.txt')
    if not any(( x in ln for x in ('ab','cd','pq','xy')))
    and sum(ln.count(c) for c in 'aeiou') >= 3
    and any( a==b for a,b in zip(ln[:-1],ln[1:]) )
)

print( "Part 1:", nice )


nice = sum( 1 for ln in open('day05.txt') 
    if any(ln[n:n+2] in ln[n+2:] for n in range(len(ln)-2))  
    and any(ln[n] == ln[n+2] for n in range(len(ln)-2))
)

print( "Part 2:", nice )

File: 2020/day05.go
package main

import (
	"Fmt"
	"bufio"
	"os"
	"sort"
)

func main() {
	file, _ := os.Open("day05.txt")
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var seats []int;
	for scanner.Scan() {
		seat := 0
                ln := scanner.Text()
		for _, c := range ln {
			seat = seat * 2
			if c == 'B' || c == 'R' {
				seat += 1
			}
		}
                seats = append( seats, seat )
	}
	sort.Ints( seats )
	fmt.Println("Part 1: ",seats[len(seats)-1])

        last := 0
        for _, seat := range seats {
            if last > 0 && seat != last+1 {
                break
            }
            last = seat
        }
        fmt.Println( "Part 2: ", last+1 )
}
